File: src/main/java/Card/Blue/DeepFreeze.java
package Card.Blue;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

public class DeepFreeze extends AbstractCard {
    public static final String ID = "Multiple Process";
    private static final CardStrings cardStrings;
    public static final String NAME;
    public static final String DESCRIPTION;
    private static final int COST = 0;

    public DeepFreeze() {
        super(ID, NAME, "1", COST, DESCRIPTION, CardType.SKILL, CardColor.BLUE, CardRarity.COMMON, CardTarget.SELF);
        this.baseMagicNumber = 3;
        this.magicNumber = baseMagicNumber;
    }

    public void use(AbstractPlayer player, AbstractMonster monster) {

    }


    public void upgrade() {
        if (!this.upgraded) {
            this.upgradeName();
            this.upgradeMagicNumber(1);
        }
    }


    public AbstractCard makeCopy() {
        return new DeepFreeze();
    }

    static {
        cardStrings = CardCrawlGame.languagePack.getCardStrings(ID);
        NAME = cardStrings.NAME;
        DESCRIPTION = cardStrings.DESCRIPTION;
    }

}

File: src/main/java/Card/Blue/DeepFreeze.kt
package Card.Blue

import Action.DeepFreezeAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster
import com.megacrit.cardcrawl.powers.AbstractPower

class DeepFreeze : AbstractCard(ID, NAME, "1", COST, DESCRIPTION, CardType.SKILL, CardColor.BLUE, CardRarity.COMMON, CardTarget.SELF) {
    init {
        this.baseMagicNumber = 3
        this.magicNumber = baseMagicNumber
    }

    override fun use(player: AbstractPlayer, monster: AbstractMonster) {
        val focus: AbstractPower =  AbstractDungeon.player.getPower("Focus")
        this.addToBot(DeepFreezeAction(monster, DamageInfo(player, focus.amount+2, DamageInfo.DamageType.NORMAL)))
        this.isUsed = true
    }

    override fun atTurnStart() {
        this.isUsed = false
        super.atTurnStart()
    }

    override fun canUse(p: AbstractPlayer?, m: AbstractMonster?): Boolean {
        return this.isUsed
    }

    override fun upgrade() {
        if (!this.upgraded) {
            this.upgradeName()
            this.upgradeMagicNumber(1)
        }
    }
    //todo 生成球时更新费用

    override fun updateCost(amt: Int) {
        super.updateCost(amt)
    }

    override fun makeCopy(): AbstractCard {
        return DeepFreeze()
    }

    companion object {
        const val ID = "DeepFreeze"
        private val cardStrings: CardStrings
        val NAME: String
        val DESCRIPTION: String
        const val COST = 0

        init {

            cardStrings = CardCrawlGame.languagePack.getCardStrings(ID)
            NAME = cardStrings.NAME
            DESCRIPTION = cardStrings.DESCRIPTION
        }
    }

}
